### PR TITLE
avoid duplicate savings

### LINKED ISSUE #206: Fix Add Trip Screen: disable "save" after trip/activity creation
Right now if you double press quickly on the save button when creating a trip/activity, the trip/activity is created twice

### PR DESCRIPTION
**Pull request by mirceagavrizi**
## Summary

This PR introduces a fix which prevents duplicate savings. Until now, if the save button was pressed multiple times quickly enough duplicate trips would've been created and in case of the activities, the user was redirected to the Overview screen.

## Changes

- **Screens/UI:** Fixes the behavior of multiple clicks on save in addActivityScreen and addTripScreen
- **Bug Fixes/Improvements:** Addresses #206 .

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #206 .
- [ ] Tests have been added for new functionality.
- [ ] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [ ] Added units tests for this

##  Related Issues/Tickets

Closes #206

### CODE REVIEW

**Review by apana03 — CHANGES_REQUESTED** [review]
Thank you for the quick bug fix! However I see that this fix is not tested, we should ensure unit tests cover the edge case of rapidly clicking the save button and verify that only one save operation is triggered.

File: app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt [hunk 1]
```diff
@@ -208,12 +211,20 @@ fun AddTripScreen(
...
 
     if (!isEditMode) {
-      tripsViewModel.createTrip(trip)
+      isSaving = true
+      tripsViewModel.createTrip(
+          trip,
+          onSuccess = {
+            isSaving = false
+            Toast.makeText(context, "Trip created successfully!", Toast.LENGTH_SHORT).show()
+          })
```
[the remark this answers is not in the document]
Comment (mirceagavrizi) [reply app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt:226]: You're right. I added the onFailure parameter for updateTrip and createTrip method in TripsViewModel and added the necessary error handlings + setting isSaving to false.

**Review by apana03 — APPROVED** [review]
LGTM! Thank you for taking into account the comments.
